multicluster/cmd/unlink.go

```diff
@@ -89,6 +89,17 @@ func newUnlinkCommand() *cobra.Command {
 				)
 			}
 
+			selector = fmt.Sprintf("%s=%s", clusterNameLabel, opts.clusterName)
+			destinationCredentials, err := k.CoreV1().Secrets(controlPlaneNamespace).List(cmd.Context(), metav1.ListOptions{LabelSelector: selector})
+			if err != nil {
+				return err
+			}
+			for _, secret := range destinationCredentials.Items {
```

Comment:
Why the looping? Isn't there gonna be just one secret for this `opts.clusterName`? If so, it could be processed in a single statement like it's done above for the other secret.

Reply:
Yeah, we could delete the secret by name but discovering it by label feels a bit more robust to me.  Labels feel like a more stable contract than names, but that may just be a vibes thing.  There should only be one secret this label, but `List()` will give us back a list so we might as well iterate over it just in case there's somehow multiple secrets with the label.